src/msg/model.rs

```diff
@@ -728,6 +732,23 @@ impl Msg {
 }
 
 // -- Traits --
+impl ser::Serialize for Msg {
```

Comment:
The reason why I implemented it manually, is because I didn't want to add another attribute for `Msg` which is only used for Serializsation.

Reply:
So I think it could be nicer to introduce a dedicated struct for this purpose, sth like:

```rust
#[derive(Serialize)]
struct SerializableMsg {
    #[serde(flatten)]
    inner: Msg,
    has_attachments: Bool,
}
```

It's better to let the `serde` lib do it's job rather than implementing ourselves Serialize.

Reply:
Btu that makes it more complicated from my point of view. Currently, we can just give `Msg` to the `Output.print` function and the method will decide if it should use the implemented serializer or the `Display` trait.

How should it be handled if you want to print out the msg? I'd need to always check which mode the printer has `Output.is_json()` and `Output.is_text()`.

Reply:
I'm not sure to understand your problem. Instead of passing a `Msg` to the printer, you pass a `SerializableMsg`. The only thing we need is a connector from a `Msg` to a `SerializableMsg`. It's quite common to have different struct from the same model for different purpose. At one of my work I used (`diesel`)[https://diesel.rs/guides/getting-started] (it's an ORM), and it's common to have a struct for the `Queryable` trait, another one for the `Insertable` trait etc.

Reply:
Could you please show my a little example what you mean? Should he `SerializeableMsg` implement the `From` trait to be used like this?
```
Output.print(SerializeableMsg::from(&msg));
```
?

Reply:
It could be great but I'm afraid you will not be able (because you will miss the `has_attachments` that does not exists on `Msg`). A custom function will be needed.